src/agent/main.cpp

```diff
@@ -77,7 +75,9 @@ static int Mainloop(AgentInstance &aInstance)
 
     while (true)
     {
-        otSysMainloopContext mainloop;
+        otSysMainloopContext                         mainloop;
+        ot::BorderRouter::Ncp::ControllerOpenThread *ncp =
```

Comment:
we are using namespace `ot::BorderRouter`, just `Ncp::Controller`. do we need use concrete `ControllerOpenThread`?